### ISSUE TITLE
安装依赖时产生了一些意想不到的问题

### ISSUE THREAD

**Issue by chu025** [sMythicalBird/ZenlessZoneZero-Auto#113]
[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: D:\juequlin\ZenlessZoneZero-Auto-master\py310\python.exe -m pip install --upgrade pip
2024/08/04 12:30:35 exit status 2
请选择需要执行的任务:
        1. 安装依赖(第一次运行时或提示缺失依赖时)
        2. 安装指定版本依赖
        3. 运行脚本
        4. 退出
请输入选择:1
请选择安装依赖版本:
        0. 智能选择（推荐）
        1. 安装 GPU 版本
        2. 安装 CPU 版本
        3. 退出
请输入选择:0
2024/08/04 12:32:31 安装依赖:requirements-cpu.txt
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting paddlepaddle (from -r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/c7/d0/a48695cfc5e452621256631ab87583268b0e3f27a91e49acf84a1ed32212/paddlepaddle-2.6.1-cp310-cp310-win_amd64.whl (81.0 MB)
Collecting paddleocr==2.7.3 (from -r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/f2/55/0469ebca1d9c581a3fa740621afe96461a0ef450e489e10e278cc17a19ef/paddleocr-2.7.3-py3-none-any.whl (780 kB)
Collecting numpy<2.0.0 (from -r requirements-cpu.txt (line 3))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/19/77/538f202862b9183f54108557bfda67e17603fc560c384559e769321c9d92/numpy-1.26.4-cp310-cp310-win_amd64.whl (15.8 MB)
Collecting loguru (from -r requirements-cpu.txt (line 4))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/03/0a/4f6fed21aa246c6b49b561ca55facacc2a44b87d65b8b92362a8e99ba202/loguru-0.7.2-py3-none-any.whl (62 kB)
Collecting opencv-python (from -r requirements-cpu.txt (line 5))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/ec/6c/fab8113424af5049f85717e8e527ca3773299a3c6b02506e66436e19874f/opencv_python-4.10.0.84-cp37-abi3-win_amd64.whl (38.8 MB)
Collecting pillow (from -r requirements-cpu.txt (line 6))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/f4/72/0203e94a91ddb4a9d5238434ae6c1ca10e610e8487036132ea9bf806ca2a/pillow-10.4.0-cp310-cp310-win_amd64.whl (2.6 MB)
Collecting pydantic>2.0 (from -r requirements-cpu.txt (line 7))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/1f/fa/b7f815b8c9ad021c07f88875b601222ef5e70619391ade4a49234d12d278/pydantic-2.8.2-py3-none-any.whl (423 kB)
Collecting requests (from -r requirements-cpu.txt (line 8))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/f9/9b/335f9764261e915ed497fcdeb11df5dfd6f7bf257d4a6a2a686d80da4d54/requests-2.32.3-py3-none-any.whl (64 kB)
Collecting tqdm (from -r requirements-cpu.txt (line 9))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/48/5d/acf5905c36149bbaec41ccf7f2b68814647347b72075ac0b1fe3022fdc73/tqdm-4.66.5-py3-none-any.whl (78 kB)
Collecting pynput (from -r requirements-cpu.txt (line 10))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/ef/1d/fdef3fdc9dc8dedc65898c8ad0e8922a914bb89c5308887e45f9aafaec36/pynput-1.7.7-py2.py3-none-any.whl (90 kB)
Collecting pydirectinput-rgx (from -r requirements-cpu.txt (line 11))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/47/b0/b684abbed1c0f1ace6bb684ee804851233fcc7077244f88347e3a42f449a/pydirectinput_rgx-2.1.1-py3-none-any.whl (35 kB)
Collecting pywin32 (from -r requirements-cpu.txt (line 12))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/d3/d6/891894edec688e72c2e308b3243fad98b4066e1839fd2fe78f04129a9d31/pywin32-306-cp310-cp310-win_amd64.whl (9.2 MB)
Collecting pyyaml (from -r requirements-cpu.txt (line 13))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/24/97/9b59b43431f98d01806b288532da38099cc6f2fea0f3d712e21e269c0279/PyYAML-6.0.1-cp310-cp310-win_amd64.whl (145 kB)
Collecting onnxruntime (from -r requirements-cpu.txt (line 14))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/74/8e/0db236760a60ba59fac48e253c2c645bfd7d6ade58d4ce796b645d73e2de/onnxruntime-1.18.1-cp310-cp310-win_amd64.whl (5.6 MB)
Collecting shapely (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/70/a5/6d171586eb850c7582d2aeab027e1027ace4eb3a6c2e5f05746b8ab039e0/shapely-2.0.5-cp310-cp310-win_amd64.whl (1.4 MB)
Collecting scikit-image (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/99/89/3fcd68d034db5d29c974e964d03deec9d0fbf9410ff0a0b95efff70947f6/scikit_image-0.24.0-cp310-cp310-win_amd64.whl (12.9 MB)
Collecting imgaug (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/66/b1/af3142c4a85cba6da9f4ebb5ff4e21e2616309552caca5e8acefe9840622/imgaug-0.4.0-py2.py3-none-any.whl (948 kB)
Collecting pyclipper (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/60/61/354f484ab7969a601327646bbaeb1b799508b4e81946ea4d52bbf9d779c6/pyclipper-1.3.0.post5-cp310-cp310-win_amd64.whl (108 kB)
Collecting lmdb (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/64/ca/5e8ed72930c410eedd25d801c8abfd6cbf65bac0461128fbdd03358b279e/lmdb-1.5.1-cp310-cp310-win_amd64.whl (100 kB)
Collecting visualdl (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/ea/b5/37726c750a4f4598660998327c3566b2d2ed5a1a5f44e9f0dde875602447/visualdl-2.5.3-py3-none-any.whl (6.3 MB)
Collecting rapidfuzz (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/8c/d7/b832df7eabfaad43ba0345695bbf3272995d607c4f0f0cba5001d8f05c0c/rapidfuzz-3.9.5-cp310-cp310-win_amd64.whl (1.7 MB)
Collecting opencv-python (from -r requirements-cpu.txt (line 5))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/cf/09/b24c266cd61ddeed101b90c92a26f54d060b06f4a1b102eb891576d6e9e2/opencv_python-4.6.0.66-cp36-abi3-win_amd64.whl (35.6 MB)
Collecting opencv-contrib-python<=4.6.0.66 (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/63/0b/6ef1acbaa21e5245c85a42f9f0ecfaf2e7420b24615a00f0eee170328e6b/opencv_contrib_python-4.6.0.66-cp36-abi3-win_amd64.whl (42.5 MB)
Collecting cython (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/83/42/9537a513d7d83196ab5973d911056eb22a639e04162fd649cf74b4dac3f1/Cython-3.0.10-cp310-cp310-win_amd64.whl (2.8 MB)
Collecting lxml (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/5b/bc/51530ee31e3ce25634ed790e38a62ffaea6507befd73b51926e19ce794f9/lxml-5.2.2-cp310-cp310-win_amd64.whl (3.8 MB)
Collecting premailer (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/b1/07/4e8d94f94c7d41ca5ddf8a9695ad87b888104e2fd41a35546c1dc9ca74ac/premailer-3.10.0-py2.py3-none-any.whl (19 kB)
Collecting openpyxl (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/c0/da/977ded879c29cbd04de313843e76868e6e13408a94ed6b987245dc7c8506/openpyxl-3.1.5-py2.py3-none-any.whl (250 kB)
Collecting attrdict (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/ef/97/28fe7e68bc7adfce67d4339756e85e9fcf3c6fd7f0c0781695352b70472c/attrdict-2.0.1-py2.py3-none-any.whl (9.9 kB)
Collecting python-docx (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/3e/3d/330d9efbdb816d3f60bf2ad92f05e1708e4a1b9abe80461ac3444c83f749/python_docx-1.1.2-py3-none-any.whl (244 kB)
Collecting beautifulsoup4 (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/b1/fe/e8c672695b37eecc5cbf43e1d0638d88d66ba3a44c4d321c796f4e59167f/beautifulsoup4-4.12.3-py3-none-any.whl (147 kB)
Collecting fonttools>=4.24.0 (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/70/11/7b81b12a5614b5d237ab70c38bdc268de3eb3880ce7bb1269122e0a415ea/fonttools-4.53.1-cp310-cp310-win_amd64.whl (2.2 MB)
Collecting fire>=0.3.0 (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/1b/1b/84c63f592ecdfbb3d77d22a8d93c9b92791e4fa35677ad71a7d6449100f8/fire-0.6.0.tar.gz (88 kB)
  Preparing metadata (setup.py) ... done
Collecting pdf2docx (from paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/b5/f9/6d567df395c0409baf2b4dd9cd30d1e977c70672fe7ec2a684af1e6aa41c/pdf2docx-0.5.8-py3-none-any.whl (132 kB)
Collecting httpx (from paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/41/7b/ddacf6dcebb42466abd03f368782142baa82e08fc0c1f8eaa05b4bae87d5/httpx-0.27.0-py3-none-any.whl (75 kB)
Collecting decorator (from paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/d5/50/83c593b07763e1161326b3b8c6686f0f4b0f24d5526546bee538c89837d6/decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting astor (from paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/c3/88/97eef84f48fa04fbd6750e62dcceafba6c63c81b7ac1420856c8dcc0a3f9/astor-0.8.1-py2.py3-none-any.whl (27 kB)
Collecting opt-einsum==3.3.0 (from paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/bc/19/404708a7e54ad2798907210462fd950c3442ea51acc8790f3da48d2bee8b/opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Collecting protobuf<=3.20.2,>=3.1.0 (from paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/39/f3/393c00e45439a46f293077da5b0362a1a4d04b2c8242c35a763f03e8e742/protobuf-3.20.2-cp310-cp310-win_amd64.whl (904 kB)
Collecting colorama>=0.3.4 (from loguru->-r requirements-cpu.txt (line 4))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/d1/d6/3965ed04c63042e047cb6a3e6ed1a63a35087b6a609aa3a15ed8ac56c221/colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting win32-setctime>=1.0.0 (from loguru->-r requirements-cpu.txt (line 4))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/0a/e6/a7d828fef907843b2a5773ebff47fb79ac0c1c88d60c0ca9530ee941e248/win32_setctime-1.1.0-py3-none-any.whl (3.6 kB)
Collecting annotated-types>=0.4.0 (from pydantic>2.0->-r requirements-cpu.txt (line 7))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/78/b6/6307fbef88d9b5ee7421e68d78a9f162e0da4900bc5f5793f6d3d0e34fb8/annotated_types-0.7.0-py3-none-any.whl (13 kB)
Collecting pydantic-core==2.20.1 (from pydantic>2.0->-r requirements-cpu.txt (line 7))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/be/4a/576524eefa9b301c088c4818dc50ff1c51a88fe29efd87ab75748ae15fd7/pydantic_core-2.20.1-cp310-none-win_amd64.whl (1.9 MB)
Collecting typing-extensions>=4.6.1 (from pydantic>2.0->-r requirements-cpu.txt (line 7))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/26/9f/ad63fc0248c5379346306f8668cda6e2e2e9c95e01216d2b8ffd9ff037d0/typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Collecting charset-normalizer<4,>=2 (from requests->-r requirements-cpu.txt (line 8))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/a2/a0/4af29e22cb5942488cf45630cbdd7cefd908768e69bdd90280842e4e8529/charset_normalizer-3.3.2-cp310-cp310-win_amd64.whl (100 kB)
Collecting idna<4,>=2.5 (from requests->-r requirements-cpu.txt (line 8))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/e5/3e/741d8c82801c347547f8a2a06aa57dbb1992be9e948df2ea0eda2c8b79e8/idna-3.7-py3-none-any.whl (66 kB)
Collecting urllib3<3,>=1.21.1 (from requests->-r requirements-cpu.txt (line 8))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/ca/1c/89ffc63a9605b583d5df2be791a27bc1a42b7c32bab68d3c8f2f73a98cd4/urllib3-2.2.2-py3-none-any.whl (121 kB)
Collecting certifi>=2017.4.17 (from requests->-r requirements-cpu.txt (line 8))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/1c/d5/c84e1a17bf61d4df64ca866a1c9a913874b4e9bdc131ec689a0ad013fb36/certifi-2024.7.4-py3-none-any.whl (162 kB)
Collecting six (from pynput->-r requirements-cpu.txt (line 10))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting coloredlogs (from onnxruntime->-r requirements-cpu.txt (line 14))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/a7/06/3d6badcf13db419e25b07041d9c7b4a2c331d3f4e7134445ec5df57714cd/coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
Collecting flatbuffers (from onnxruntime->-r requirements-cpu.txt (line 14))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/41/f0/7e988a019bc54b2dbd0ad4182ef2d53488bb02e58694cd79d61369e85900/flatbuffers-24.3.25-py2.py3-none-any.whl (26 kB)
Collecting packaging (from onnxruntime->-r requirements-cpu.txt (line 14))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/08/aa/cc0199a5f0ad350994d660967a8efb233fe0416e4639146c089643407ce6/packaging-24.1-py3-none-any.whl (53 kB)
Collecting sympy (from onnxruntime->-r requirements-cpu.txt (line 14))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/b2/fe/81695a1aa331a842b582453b605175f419fe8540355886031328089d840a/sympy-1.13.1-py3-none-any.whl (6.2 MB)
Collecting termcolor (from fire>=0.3.0->paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/d9/5f/8c716e47b3a50cbd7c146f45881e11d9414def768b7cd9c5e6650ec2a80a/termcolor-2.4.0-py3-none-any.whl (7.7 kB)
Collecting soupsieve>1.2 (from beautifulsoup4->paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/4c/f3/038b302fdfbe3be7da016777069f26ceefe11a681055ea1f7817546508e3/soupsieve-2.5-py3-none-any.whl (36 kB)
Collecting humanfriendly>=9.1 (from coloredlogs->onnxruntime->-r requirements-cpu.txt (line 14))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/f0/0f/310fb31e39e2d734ccaa2c0fb981ee41f7bd5056ce9bc29b2248bd569169/humanfriendly-10.0-py2.py3-none-any.whl (86 kB)
Collecting anyio (from httpx->paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/7b/a2/10639a79341f6c019dedc95bd48a4928eed9f1d1197f4c04f546fc7ae0ff/anyio-4.4.0-py3-none-any.whl (86 kB)
Collecting httpcore==1.* (from httpx->paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/78/d4/e5d7e4f2174f8a4d63c8897d79eb8fe2503f7ecc03282fee1fa2719c2704/httpcore-1.0.5-py3-none-any.whl (77 kB)
Collecting sniffio (from httpx->paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/e9/44/75a9c9421471a6c4805dbf2356f7c181a29c1879239abab1ea2cc8f38b40/sniffio-1.3.1-py3-none-any.whl (10 kB)
Collecting h11<0.15,>=0.13 (from httpcore==1.*->httpx->paddlepaddle->-r requirements-cpu.txt (line 1))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/95/04/ff642e65ad6b90db43e668d70ffb6736436c7ce41fcc549f4e9472234127/h11-0.14.0-py3-none-any.whl (58 kB)
Collecting scipy (from imgaug->paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/a3/0c/82c1330c08f31d61142d38cb9a185e01c2403c990d10dab208032e62d0fa/scipy-1.14.0-cp310-cp310-win_amd64.whl (44.8 MB)
Collecting matplotlib (from imgaug->paddleocr==2.7.3->-r requirements-cpu.txt (line 2))
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/22/06/9e8ba6ec8b716a215404a5d1938b61f5a28001be493cf35344dda9a4072a/matplotlib-3.9.1.tar.gz (36.1 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
**ERROR: Exception:
Traceback (most recent call last):
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\cli\base_command.py", line 180, in exc_logging_wrapper
    status = run_func(*args)
  File "D:\juequlin\ZenlessZoneZero-Auto-maste**r\py310\lib\site-packages\pip\_internal\cli\req_command.py", line 245, in wrapper
    return func(self, options, args)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\commands\install.py", line 377, in run
    requirement_set = resolver.resolve(
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\resolver.py", line 95, in resolve
    result = self._result = resolver.resolve(
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\resolvelib\resolvers.py", line 546, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\resolvelib\resolvers.py", line 427, in resolve
    failure_causes = self._attempt_to_pin_criterion(name)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\resolvelib\resolvers.py", line 239, in _attempt_to_pin_criterion
    criteria = self._get_updated_criteria(candidate)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\resolvelib\resolvers.py", line 230, in _get_updated_criteria
    self._add_to_criteria(criteria, requirement, parent=candidate)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\resolvelib\resolvers.py", line 173, in _add_to_criteria
    if not criterion.candidates:
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\resolvelib\structs.py", line 156, in __bool__
    return bool(self._sequence)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\found_candidates.py", line 155, in __bool__
    return any(self)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\found_candidates.py", line 143, in <genexpr>
    return (c for c in iterator if id(c) not in self._incompatible_ids)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\found_candidates.py", line 47, in _iter_built
    candidate = func()
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\factory.py", line 182, in _make_candidate_from_link
    base: Optional[BaseCandidate] = self._make_base_candidate_from_link(
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\factory.py", line 228, in _make_base_candidate_from_link
    self._link_candidate_cache[link] = LinkCandidate(
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 290, in __init__
    super().__init__(
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 156, in __init__
    self.dist = self._prepare()
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 222, in _prepare
    dist = self._prepare_distribution()
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 301, in _prepare_distribution
    return preparer.prepare_linked_requirement(self._ireq, parallel_builds=True)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\operations\prepare.py", line 525, in prepare_linked_requirement
    return self._prepare_linked_requirement(req, parallel_builds)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\operations\prepare.py", line 640, in _prepare_linked_requirement
    dist = _get_prepared_distribution(
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\operations\prepare.py", line 71, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\distributions\sdist.py", line 54, in prepare_distribution_metadata
    self._install_build_reqs(finder)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\distributions\sdist.py", line 124, in _install_build_reqs
    build_reqs = self._get_build_requires_wheel()
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\distributions\sdist.py", line 101, in _get_build_requires_wheel
    return backend.get_requires_for_build_wheel()
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_internal\utils\misc.py", line 745, in get_requires_for_build_wheel
    return super().get_requires_for_build_wheel(config_settings=cs)
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\pyproject_hooks\_impl.py", line 166, in get_requires_for_build_wheel
    return self._call_hook('get_requires_for_build_wheel', {
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\pyproject_hooks\_impl.py", line 321, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pip._vendor.pyproject_hooks._impl.BackendUnavailable: Traceback (most recent call last):
  File "D:\juequlin\ZenlessZoneZero-Auto-master\py310\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 77, in _build_backend
    obj = import_module(mod_path)
  File "importlib\__init__.py", line 126, in import_module
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'mesonpy'
![image](https://github.com/user-attachments/assets/5891fcd0-28c1-44fd-975e-68a293dc76e5)

[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: D:\juequlin\ZenlessZoneZero-Auto-master\py310\python.exe -m pip install --upgrade pip
2024/08/04 12:34:11 exit status 2
请选择需要执行的任务:
        1. 安装依赖(第一次运行时或提示缺失依赖时)
        2. 安装指定版本依赖
        3. 运行脚本
        4. 退出
请输入选择:3

 --------------------------------------------------------------
     注意：此脚本为免费的开源软件，如果你是通过购买获得的，那么你受骗了！
--------------------------------------------------------------

使用说明：
   F10  恢复运行
   F11  暂停运行
   F12  结束运行

Traceback (most recent call last):
  File "D:\juequlin\ZenlessZoneZero-Auto-master\main.py", line 30, in <module>
    from pynput.keyboard import Key, Listener
ModuleNotFoundError: No module named 'pynput'
2024/08/04 12:34:18 exit status 1
请选择需要执行的任务:
        1. 安装依赖(第一次运行时或提示缺失依赖时)
        2. 安装指定版本依赖
        3. 运行脚本
        4. 退出
请输入选择:

**Comment by sMythicalBird** [comment]
py310删了重装

**Comment by whx1024** [comment]
这么解决，编辑requirements.txt，指定matplotlib==3.9.0，并在paddleocr上面。例如：paddlepaddle
matplotlib==3.9.0
paddleocr==2.7.3。这样就能解决问题。原因：paddleocr构建时需matplotlib，但最新版3.9.1可能更新了某些依赖，导致本地构建失败。

**Comment by tacaswell** [comment]
The 3.9.1 wheels on windows would, depending on import order, cause segfaults, see https://github.com/matplotlib/matplotlib/issues/28551#issuecomment-2266699619.

We deleted the bad wheels to avoid segfaults (which seemed like a good idea) but failed to anticipate that this would cause downstream projects to start trying to build mpl from source instead of pulling the 3.9.0 wheel.  We have now yanked the whole 3.9.1 release to fix these problems and will get 3.9.2 out as quick as we can.

To prevent this sort of thing in the future (trying to build something from source when you are not prepared to do so) I suggest using `matplotlib --only-binary "matplotlib" ` in your requirements file (which may be a good practice for all wheels that have extensions).

Sorry for the trouble.

----

I apologize for responding in English, but what little 中文 I once knew I no longer can use.
